3scaleAdapter/pkg/threescale/threescale.go

```diff
@@ -64,73 +64,74 @@ var _ logentry.HandleLogEntryServiceServer = &Threescale{}
 
 func (s *Threescale) HandleLogEntry(ctx context.Context, l *logentry.HandleLogEntryRequest) (*v1beta1.ReportResult, error) {
 	var report v1beta1.ReportResult
-	var err error
 
-	if l.AdapterConfig == nil {
-		return &report, fmt.Errorf("adapter config not available")
-	}
+	go func() {
+		var err error
 
-	cfg := &config.Params{}
-	if err := cfg.Unmarshal(l.AdapterConfig.Value); err != nil {
-		log.Errorf("error unmarshalling adapter config: %v", err)
-		return &report, err
-	}
+		if l.AdapterConfig == nil {
+			err = errors.New("adapter config not available")
+		}
```

Comment:
since AdapterConfig is nil in this case, shouldn't we just finish this goroutine (and log the error)?

Reply:
as pointed on the previous comment.. yes, we should log and finish there.

Reply:
:+1: